### PR TITLE
zfs mount man page should document legacy behaviour

### LINKED ISSUE #2900: Add information about what happens to child filesystems when invoking mount on the parent filesystem to man zfs
It is currently not as clear as it could be from the `zfs mount` section in `man zfs` what happens to a filesystem x/y when `zfs mount x` is invoked (i.e. whether it is mounted or not when mountpoint is `legacy` or not).

### PR DESCRIPTION
**Pull request by kpande**
Closes #2900

Signed-off-by: Kash Pande <kash@tripleback.net>

### Motivation and Context
Related-to: #2900

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: man/man8/zfs.8 [hunk 1]
```diff
@@ -3304,7 +3304,7 @@ Displays all ZFS file systems currently mounted.
 .Op Fl o Ar options
 .Fl a | Ar filesystem
 .Xc
-Mounts ZFS file systems.
+Mount ZFS filesystem if a non-legacy mountpoint is set.
```
Comment (ahrens) [inline man/man8/zfs.8]: I think we could be even more specific by saying, `Mount ZFS filesystem on the directory specified by the 'mountpoint' property.  However, if the mountpoint property is 'legacy', the mount request is ignored and the filesystem is not mounted.`

**Comment by kpande** [comment]
alright, fixed the text with overlay mount to explain what that's about, too.

File: man/man8/zfs.8 [hunk 2]
```diff
@@ -3304,16 +3304,24 @@ Displays all ZFS file systems currently mounted.
 .Op Fl o Ar options
 .Fl a | Ar filesystem
 .Xc
-Mounts ZFS file systems.
+Mount ZFS filesystem on its configured
+.Sy mountpoint
```
Comment (allanjude) [inline man/man8/zfs.8]: because 'mountpoint' is a generic term, I think we should be specific here, that we are talking about the mountpoint property.

**Comment by kpande** [comment]
made sure it mentions that it is the mountpoint property.

also saw that I stupidly said "non-empty" when it should have said "empty".

**Pull request merged**